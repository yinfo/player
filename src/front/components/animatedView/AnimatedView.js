// @flow

import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import { withRouter } from 'react-router-dom';

type Props = {
  children: any,
  animated: boolean,
};

function AnimatedView({ animated = true, children }: Props) {
  return (
    <section
      className={cx({
        content: true,
        'view-enter': animated,
      })}
    >
      {children}
    </section>
  );
}

AnimatedView.displayName = 'AnimatedView';

AnimatedView.propTypes = {
  children: PropTypes.node,
  animated: PropTypes.bool,
};

export default withRouter(AnimatedView);
