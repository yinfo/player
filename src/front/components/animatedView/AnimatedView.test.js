// @flow

import React from 'react';
import { shallow, mount } from 'enzyme';
import { MemoryRouter } from 'react-router';
import AnimatedView from './AnimatedView';

describe('AnimatedView component', () => {
  it('renders as expected', () => {
    const component = shallow(
      <MemoryRouter>
        <AnimatedView>
          <p>an AnimatedView</p>
        </AnimatedView>
      </MemoryRouter>,
    );

    expect(component).toMatchSnapshot();
  });

  it('should be animated by default', () => {
    const component = mount(
      <MemoryRouter>
        <AnimatedView>
          <p>an AnimatedView</p>
        </AnimatedView>
      </MemoryRouter>,
    );

    expect(component.find('.content').hasClass('view-enter')).toEqual(true);
  });

  it('should NOT be animated when prop animated is false', () => {
    const component = mount(
      <MemoryRouter>
        <AnimatedView animated={false}>
          <p>an AnimatedView</p>
        </AnimatedView>
      </MemoryRouter>,
    );

    expect(component.find('.content').hasClass('view-enter')).toEqual(false);
  });
});
