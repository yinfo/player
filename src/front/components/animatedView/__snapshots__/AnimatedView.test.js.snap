// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimatedView component renders as expected 1`] = `ShallowWrapper {}`;
